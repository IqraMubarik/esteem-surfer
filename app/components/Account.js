/*
eslint-disable react/no-multi-comp
*/


import React, {Component, Fragment} from 'react';

import PropTypes from 'prop-types';

import {Tooltip} from 'antd';

import {FormattedNumber, FormattedDate} from 'react-intl';

import NavBar from './layout/NavBar';

import ComposeBtn from './elements/ComposeBtn';
import UserAvatar from './elements/UserAvatar';
import FollowControls from './elements/FollowControls';

import {getFollowCount, getAccount} from '../backend/steem-client';

import {getActiveVotes, getTopPosts} from '../backend/esteem-client';

import authorReputation from '../utils/author-reputation';
import {votingPower} from '../utils/manabar';
import proxifyImageSrc from '../utils/proxify-image-src';
import {makeGroupKeyForEntries} from "../actions/entries";
import EntryListLoadingItem from "./elements/EntryListLoadingItem";
import EntryListItem from "./elements/EntryListItem";
import AppFooter from "./layout/AppFooter";
import ScrollReplace from "./helpers/ScrollReplace";
import ListSwitch from "./elements/ListSwitch";
import coverFallbackDay from '../img/cover-fallback-day.png';
import coverFallbackNight from '../img/cover-fallback-night.png';
import LinearProgress from "./common/LinearProgress";

import catchEntryImage from '../utils/catch-entry-image';
import entryBodySummary from "../utils/entry-body-summary";

class Profile extends Component {

  render() {

    let vPower;
    let vPowerPercentage;
    let reputation;
    let name;
    let about;
    let postCount;
    let activeVotes;
    let followerCount;
    let followingCount;
    let location;
    let website;
    let created;

    const {username, account} = this.props;

    if (account) {
      vPower = votingPower(account);
      vPowerPercentage = `${parseInt(vPower, 10)}%`;
      reputation = authorReputation(account.reputation);
      postCount = account.post_count;
      ({activeVotes} = account);
      ({followerCount} = account);
      ({followingCount} = account);

      const {accountProfile} = account;
      if (accountProfile) {
        name = accountProfile.name || null;
        about = accountProfile.about || null;
        location = accountProfile.location || null;
        website = accountProfile.website || null;
      }

      created = new Date(account.created);
    }

    return (
      <div className="profile-area">
        <div className="account-avatar">
          <UserAvatar user={username} size="xLarge"/>
          {reputation && <div className="reputation">{reputation}</div>}
        </div>

        <div className="username">{username}</div>

        {vPowerPercentage && (
          <div className="vpower-line">
            <div
              className="vpower-line-inner"
              style={{width: vPowerPercentage}}
            />
          </div>
        )}

        {vPower && (
          <div className="vpower-percentage">
            <Tooltip title="Voting Power">
              {vPower.toFixed(2)}
            </Tooltip>
          </div>
        )}

        {name && <div className="full-name">{name}</div>}

        {about && <div className="about">{about}</div>}

        {(name || about) && <div className="divider"/>}

        <div className="account-numbers">
          <div className="account-prop">
            <Tooltip title="Post Count" className="holder-tooltip">
              <i className="mi">list</i>
              {typeof postCount === 'number' ? (
                <FormattedNumber value={postCount}/>
              ) : (
                <span>--</span>
              )}
            </Tooltip>
          </div>
          <div className="account-prop">
            <Tooltip
              title="Number of votes in last 24 hours"
              className="holder-tooltip"
            >
              <i className="mi active-votes-icon">keyboard_arrow_up</i>
              {typeof activeVotes === 'number' ? (
                <FormattedNumber value={activeVotes}/>
              ) : (
                <span>--</span>
              )}
            </Tooltip>
          </div>
          <div className="account-prop">
            <Tooltip title="Followers" className="holder-tooltip">
              <i className="mi">people</i>
              {typeof followerCount === 'number' ? (
                <FormattedNumber value={followerCount}/>
              ) : (
                <span>--</span>
              )}
            </Tooltip>
          </div>
          <div className="account-prop">
            <Tooltip title="Following" className="holder-tooltip">
              <i className="mi">person_add</i>
              {typeof followingCount === 'number' ? (
                <FormattedNumber value={followingCount}/>
              ) : (
                <span>--</span>
              )}
            </Tooltip>
          </div>
        </div>

        <div className="divider"/>

        {location && (
          <div className="account-prop">
            <i className="mi">near_me</i> {location}
          </div>
        )}

        {website && (
          <div className="account-prop prop-website">
            <i className="mi">public</i>{' '}
            <a className="website-link">{website}</a>
          </div>
        )}

        {created && (
          <div className="account-prop">
            <i className="mi">date_range</i>{' '}
            <FormattedDate
              month="long"
              day="2-digit"
              year="numeric"
              value={created}
            />
          </div>
        )}
      </div>
    )
  }
}


Profile.defaultProps = {
  account: null,
};

Profile.propTypes = {
  username: PropTypes.string.isRequired,
  account: PropTypes.instanceOf(Object)
};


export class AccountMenu extends Component {
  goSection = (section) => {
    const {history, username} = this.props;
    const u = section ? `/@${username}/${section}` : `/@${username}`;
    history.push(u);
  };

  render() {
    const {section} = this.props;

    return (
      <div className="account-menu">
        <div className="account-menu-items">
          <a role="none" className={`menu-item ${section === 'blog' && 'selected-item'}`} onClick={() => {
            this.goSection('blog')
          }}>Blog</a>
          <a role="none" className={`menu-item ${section === 'comments' && 'selected-item'}`} onClick={() => {
            this.goSection('comments')
          }}>Comments</a>
          <a role="none" className={`menu-item ${section === 'replies' && 'selected-item'}`} onClick={() => {
            this.goSection('replies')
          }}>Replies</a>
          <a role="none" className={`menu-item ${section === 'wallet' && 'selected-item'}`} onClick={() => {
            this.goSection('wallet')
          }}>Wallet</a>

        </div>

        <div className="page-tools">
          <ListSwitch {...this.props} />
        </div>
      </div>
    )
  }
}

AccountMenu.propTypes = {
  username: PropTypes.string.isRequired,
  section: PropTypes.string.isRequired,
  history: PropTypes.instanceOf(Object).isRequired
};

export class AccountCover extends Component {
  render() {
    let coverImage;

    const {account, username, global} = this.props;

    if (account) {
      const {accountProfile} = account;
      if (accountProfile) {
        coverImage = accountProfile.cover_image || null;
      }
    }

    const bgImage = coverImage && proxifyImageSrc(coverImage) || (global.theme === 'day' ? coverFallbackDay : coverFallbackNight);

    return <div className="account-cover">
      <div className="cover-image" style={{backgroundImage: `url('${bgImage}')`}}/>
      <div className="follow-controls-holder">
        <FollowControls {...this.props} targetUsername={username}/>
      </div>
    </div>
  }
}

AccountCover.defaultProps = {
  account: null,
};

AccountCover.propTypes = {
  username: PropTypes.string.isRequired,
  account: PropTypes.instanceOf(Object),
  global: PropTypes.shape({
    theme: PropTypes.string.isRequired
  }).isRequired
};

export class SectionBlog extends Component {


}

export class SectionComments extends Component {

}

export class SectionReplies extends Component {

}

export class SectionWallet extends Component {

}

class Account extends Component {
  constructor(props) {
    super(props);

    this.state = {
      account: null,
      topPosts: null
    };
  }

  componentDidMount() {
    const {match} = this.props;
    const {username} = match.params;

    // check user here

    this.fetchAccount();
    this.fetchEntries();
    this.fetchTopPosts();
  }

  componentDidUpdate(prevProps) {
    const {location} = this.props;

    if (location !== prevProps.location) {
      this.fetchEntries();

      const {match: newMatch} = this.props;
      const {match: oldMatch} = prevProps;

      const {username: newUsername} = newMatch.params;
      const {username: oldUsername} = oldMatch.params;

      if (newUsername !== oldUsername) {
        this.fetchAccount();
        this.fetchTopPosts();
      }
    }
  }

  fetchAccount = async () => {
    const {match} = this.props;
    const {username} = match.params;

    let {visitingAccount: account} = this.props;

    if (!(account && account.name === username)) {
      account = await getAccount(username);
    }

    // Profile data
    let accountProfile;
    try {
      accountProfile = JSON.parse(account.json_metadata).profile;
    } catch (err) {
      accountProfile = null;
    }

    account = Object.assign({}, account, {accountProfile});
    this.setState({account});

    // Follow counts
    let follow;
    try {
      follow = await getFollowCount(username);
    } catch (err) {
      follow = null;
    }

    if (follow) {
      const followerCount = follow.follower_count;
      const followingCount = follow.following_count;

      account = Object.assign({}, account, {followerCount, followingCount});
      this.setState({account});
    }

    // Active votes
    let activeVotes;
    try {
      activeVotes = await getActiveVotes(username);
    } catch (err) {
      activeVotes = {count: 0};
    }

    account = Object.assign({}, account, {activeVotes: activeVotes.count});
    this.setState({account});
  };

  fetchEntries = () => {
    const {actions, match} = this.props;
    const {username, section = 'blog'} = match.params;
    actions.fetchEntries(section, `@${username}`);
  };

  fetchTopPosts = async () => {
    const {match} = this.props;
    const {username} = match.params;

    let topPosts;
    try {
      const resp = await getTopPosts(username);
      topPosts = resp.list;
    } catch (err) {
      topPosts = null;
    }

    this.setState({topPosts});
  };

  bottomReached = () => {
    const {actions, entries, match} = this.props;
    const {username, section = 'blog'} = match.params;

    const groupKey = makeGroupKeyForEntries(section, `@${username}`);
    const data = entries.get(groupKey);
    const loading = data.get('loading');
    const hasMore = data.get('hasMore');

    if (!loading && hasMore) {
      actions.fetchEntries(section, `@${username}`, true);
    }
  };

  refresh = () => {
    const {actions, match} = this.props;
    const {username, section = 'blog'} = match.params;

    this.fetchAccount();
    this.fetchTopPosts();
    actions.invalidateEntries(section, `@${username}`);
    actions.fetchEntries(section, `@${username}`, false);

    document.querySelector('#app-content').scrollTop = 0;
  };

  render() {
    const {entries, global, match} = this.props;
    const {account} = this.state;
    const {username, section = 'blog'} = match.params;
    const isWallet = section === 'wallet';

    const groupKey = makeGroupKeyForEntries(section, `@${username}`);

    const data = entries.get(groupKey);
    const entryList = data.get('entries');
    const loading = data.get('loading');

    const {topPosts} = this.state;


    return (
      <div className="wrapper">
        <NavBar
          {...this.props}
          reloadFn={() => {
            this.refresh();
          }}
          reloading={loading}
          favoriteFn={() => {
          }}
        />

        <div className="app-content account-page">
          <div className="page-header">
            <div className="left-side">
              <ComposeBtn {...this.props} />
            </div>
            <div className="right-side">
              <AccountMenu {...this.props} section={section} username={username}/>
            </div>
          </div>
          <div className="page-inner" id="app-content">
            <div className="left-side">
              <Profile {...this.props} username={username} account={account}/>
            </div>

            <div className="right-side">
              {!isWallet &&
              <AccountCover {...this.props} account={account} username={username}/>
              }

              {section === 'blog' && topPosts &&
              <div className="top-posts-list">
                <h2 className="top-posts-list-header">Top Posts</h2>

                <div className="top-posts-list-body">
                  {topPosts.map(p => (
                    <div className="top-posts-list-item" key={p.permlink}>
                      <div className="post-image">
                        <img alt="" src={catchEntryImage(p) || 'img/noimage.png'}/>
                      </div>

                      <div className="post-content">
                        <div className="post-title">{p.title}</div>
                        <div className="post-body">{entryBodySummary(p.body, 40)}</div>
                      </div>

                    </div>
                  ))}
                </div>
              </div>
              }

              {!isWallet &&
              <Fragment>
                <div className={`entry-list ${loading ? 'loading' : ''}`}>
                  <div
                    className={`entry-list-body ${
                      global.listStyle === 'grid' ? 'grid-view' : ''
                      }`}
                  >
                    {loading && entryList.size === 0 ? (
                      <EntryListLoadingItem/>
                    ) : (
                      ''
                    )}
                    {entryList.valueSeq().map(d => (
                      <EntryListItem key={d.id} {...this.props} entry={d} asAuthor={username}/>
                    ))}
                  </div>
                </div>
                {loading && entryList.size > 0 ? <LinearProgress/> : ''}
                <ScrollReplace {...this.props} selector="#app-content" onBottom={this.bottomReached}/>
              </Fragment>
              }
            </div>
          </div>
        </div>
        <AppFooter {...this.props} />

      </div>
    );
  }
}

Account.defaultProps = {
  visitingAccount: null,
  activeAccount: null
};

Account.propTypes = {
  actions: PropTypes.shape({
    fetchEntries: PropTypes.func.isRequired,
    invalidateEntries: PropTypes.func.isRequired,
    changeTheme: PropTypes.func.isRequired,
    changeListStyle: PropTypes.func.isRequired
  }).isRequired,
  global: PropTypes.shape({
    listStyle: PropTypes.string.isRequired
  }).isRequired,
  entries: PropTypes.instanceOf(Object).isRequired,
  location: PropTypes.instanceOf(Object).isRequired,
  history: PropTypes.instanceOf(Object).isRequired,
  match: PropTypes.instanceOf(Object).isRequired,
  visitingAccount: PropTypes.instanceOf(Object),
  activeAccount: PropTypes.instanceOf(Object)
};

export default Account;
