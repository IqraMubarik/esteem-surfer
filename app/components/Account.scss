@import '../style/mixins';
@import '../style/colors';

.account-page {
  .page-header {
    display: flex;
    flex-wrap: wrap;
    height: 50px;

    .left-side {
      height: 50px;
      margin-right: 32px;
      width: 208px;
    }

    .right-side {
      flex-grow: 1;

      .account-menu {
        display: flex;
        height: 50px;

        @include themify(day) {
          border-bottom: 1px solid $white-four;
        }

        @include themify(night) {
          border-bottom: 1px solid $dark-two;
        }

        .account-menu-items {
          display: flex;
          flex-shrink: 0;

          .menu-item {
            align-items: center;
            color: $steel-grey;
            display: flex;
            font-size: 16px;
            height: 50px;
            margin-right: 16px;
            padding: 0 16px;

            &.selected-item {
              border-bottom: 2px solid $dark-sky-blue;
              color: $dark-sky-blue;
              font-weight: 700;
              pointer-events: none;
            }
          }
        }

        .page-tools {
          align-items: center;
          display: flex;
          flex-grow: 1;
          justify-content: flex-end;
          padding-right: 18px;
        }
      }
    }
  }

  .page-inner {
    height: calc(100% - 50px);
    overflow-y: auto;

    .left-side {
      float: left;
      margin-right: 32px;
      width: 208px;

      .profile-area {
        padding: 16px 0 0 29px;

        .account-avatar {
          height: 128px;
          margin: 0 0 24px 0;
          position: relative;
          width: 128px;

          .user-avatar {
            border: 3px solid $duck-egg-blue;
            height: 128px;
            width: 128px;
          }

          .reputation {
            background: $dark-sky-blue;
            border-radius: 25px;
            bottom: 2px;
            color: $white;
            font-weight: 700;
            left: 0;
            line-height: 1em;
            padding: 4px 12px;
            position: absolute;
          }
        }

        .username {
          font-size: 18px;
          font-weight: 700;
          margin-bottom: 12px;

          @include themify(day) {
            color: $charcoal-grey;
          }
          @include themify(night) {
            color: $pinkish-grey;
          }

          &:before {
            content: '@';
          }
        }

        .vpower-line {
          background: $duck-egg-blue;
          height: 2px;
          margin-bottom: 8px;

          .vpower-line-inner {
            background: $dark-sky-blue;
            height: 2px;
          }
        }

        .vpower-percentage {
          color: $dark-sky-blue;
          cursor: default;
          flex-shrink: 0;
          font-size: 12px;
          font-weight: 500;
          margin-bottom: 22px;

          &:after {
            content: '%';
          }
        }

        .full-name {
          color: $dark-sky-blue;
          font-size: 18px;
          font-weight: 700;
          margin-bottom: 12px;
        }

        .about {
          color: $dark-sky-blue;
          margin-bottom: 24px;
          opacity: 0.5;
        }

        .account-prop {
          align-items: start;
          color: $steel-grey;
          cursor: default;
          display: flex;
          margin-bottom: 12px;
          word-break: break-all;

          .holder-tooltip {
            display: flex;
          }

          .mi {
            color: $silver;
            font-size: 16px;
            margin-right: 6px;
          }

          .active-votes-icon {
            align-items: center;
            background: $silver;
            border-radius: 50%;
            display: flex;
            height: 16px;
            justify-content: center;
            margin-left: 1px;
            width: 16px;

            @include themify(day) {
              color: $white;
            }

            @include themify(night) {
              color: $charcoal-grey;
            }
          }
        }

        .account-numbers {
          align-items: flex-start;
          display: flex;
          flex-wrap: wrap;
          margin-bottom: 12px;

          .account-prop {
            width: 88px;
          }
        }

        .website-link {
          color: $steel-grey;
          overflow: hidden;
          text-overflow: ellipsis;
          white-space: nowrap;
          width: 180px;
        }

        .divider {
          height: 1px;
          margin-bottom: 24px;
          overflow: hidden;

          @include themify(day) {
            background: $white-four;
          }

          @include themify(night) {
            background: $dark-two;
          }
        }
      }
    }

    .right-side {
      float: right;
      margin-top: 12px;
      width: calc(100% - 240px);

      .account-cover {
        height: 134px;
        position: relative;
        width: calc(100% - 18px);

        .cover-image {
          background-color: $white-three;
          background-position: 50% 50%;
          background-repeat: no-repeat;
          background-size: cover;
          border-radius: 3px;
          height: 100%;
          position: absolute;
          width: 100%;
          z-index: 1;
        }

        .follow-controls-holder {
          position: absolute;
          right: 10px;
          top: 10px;
          z-index: 2;
        }
      }

      .top-posts-list {
        margin: 25px 0 29px 0;
        width: calc(100% - 18px);

        .top-posts-list-header {
          font-size: 18px;
          font-weight: 700;
          margin: 0 0 16px 0;
        }

        .top-posts-list-body {
          display: flex;
          justify-content: space-between;
          max-width: 750px;
          .top-posts-list-item {
            border: 1px solid $white-four;
            border-radius: 8px;
            height: 120px;
            width: 130px;
            cursor: pointer;

            .post-image {
              border-top-left-radius: 8px;
              border-top-right-radius: 8px;
              height: 77px;
              max-width: 100%;
              overflow: hidden;

              img {
                width: 100%;
              }
            }

            .post-content {
              align-items: center;
              display: flex;
              flex-direction: column;
              height: 38px;
              justify-content: space-evenly;
              margin-top: 2px;

              .post-title,
              .post-body {
                font-size: 11px;
                overflow: hidden;
                text-overflow: ellipsis;
                white-space: nowrap;
                width: 90%;
              }

              .post-title {
                color: $charcoal-grey;
                font-weight: 700;
              }

              .post-body {
                color: $steel-grey;
              }
            }
          }
        }
      }

      .entry-list {
        .entry-list-body.grid-view {
          align-content: stretch;
          display: flex;
          flex-wrap: wrap;
          justify-content: flex-start;
          padding-right: 10px;
        }
      }

      .wallet-section {
        cursor: default;

        .first-row {
          align-items: center;
          box-sizing: content-box;
          display: flex;
          height: 50px;
          padding: 4px 0 14px 0;

          @include themify(day) {
            border-bottom: 1px solid $white-four;
          }

          @include themify(night) {
            border-bottom: 1px solid $dark-two;
          }

          .unclaimed-rewards {
            align-items: center;
            display: flex;

            .title {
              font-size: 18px;
              font-weight: 700;
              margin-right: 17px;

              @include themify(day) {
                color: $steel-grey;
              }

              @include themify(night) {
                color: $metallic-blue;
              }
            }

            .rewards {
              background-color: $dark-sky-blue;
              border-radius: 25px;
              color: $white;
              font-weight: 700;
              padding: 17px 24px;

              .reward-type {
                margin-right: 5px;
              }
            }
          }

          .estimated-value {
            color: $dark-sky-blue;
            flex-grow: 1;
            font-size: 24px;
            font-weight: 700;
            margin-right: 29px;
            text-align: right;
          }
        }

        .second-row {
          display: flex;
          justify-content: space-between;
          margin-bottom: 16px;
          padding-bottom: 16px;

          @include themify(day) {
            border-bottom: 1px solid $white-four;
          }

          @include themify(night) {
            border-bottom: 1px solid $dark-two;
          }

          .funds {
            width: calc(50% - 16px);

            .fund-info-icon {
              width: 30px;

              &:after {
                align-items: center;

                border-radius: 50%;
                color: $steel-grey;
                content: 'i';
                display: flex;
                font-size: 12px;
                height: 16px;
                justify-content: center;
                opacity: 0.5;
                width: 16px;

                @include themify(day) {
                  border: 2px solid $steel-grey;
                  color: $steel-grey;
                }

                @include themify(night) {
                  border: 2px solid $metallic-blue;
                  color: $metallic-blue;
                }
              }
            }

            .fund {
              border-radius: 8px;
              box-sizing: content-box;
              padding: 16px 16px 0 16px;

              &.alternative {
                @include themify(day) {
                  background: $white-three;
                }

                @include themify(night) {
                  background: lighten($dark-six, 4);
                }
              }

              .fund-line {
                align-items: center;
                display: flex;
                padding-bottom: 16px;

                .fund-title {
                  font-size: 16px;
                  font-weight: 500;
                  width: 110px;

                  @include themify(day) {
                    color: $charcoal-grey;
                  }

                  @include themify(night) {
                    color: $powder-blue;
                  }
                }

                .fund-number {
                  color: $dark-sky-blue;
                  flex-grow: 1;
                  font-size: 14px;
                  font-weight: 500;
                  text-align: right;
                  width: 180px;

                  &.delegated-shares,
                  &.received-shares {
                    color: $steel-grey;
                  }

                  &.total-sp {
                    font-weight: 700;
                  }
                }

                .fund-action {
                  width: 40px;
                }
              }
            }

            .next-power-down {
              align-items: center;
              box-sizing: content-box;
              display: flex;
              padding: 16px 16px 0 16px;

              @include themify(day) {
                color: $steel-grey;
              }

              @include themify(night) {
                color: $metallic-blue;
              }
            }
          }

          .exchange-data {
            margin-right: 20px;
            padding-top: 16px;
            width: calc(50% - 36px);

            .exchange-title {
              align-items: center;
              display: flex;
              margin-bottom: 14px;

              .title-icon {
                border-radius: 2px;
                font-size: 16px;
                height: 16px;
                margin-left: 16px;
                margin-right: 12px;

                @include themify(day) {
                  background: $silver;
                  color: $white;
                }

                @include themify(night) {
                  background: $metallic-blue;
                  color: $powder-blue;
                }
              }

              h2 {
                font-size: 18px;
                margin: 0;
              }
            }

            .market-data {
              border-radius: 8px;
              margin-bottom: 12px;
              padding: 14px 16px;

              @include themify(day) {
                background: $white;
                border: 1px solid $white-four;
              }

              @include themify(night) {
                background: lighten($dark-six, 4);
              }

              .market-data-title {
                font-size: 16px;
                font-weight: 500;
                margin-bottom: 14px;

                @include themify(day) {
                  color: $charcoal-grey;
                }

                @include themify(night) {
                  color: $powder-blue;
                }
              }

              .price {
                align-items: center;
                display: flex;
                margin-top: 14px;

                .price-usd {
                  color: $dark-sky-blue;
                  font-size: 18px;
                  font-weight: 500;
                }

                .price-btc {
                  color: $silver;
                  font-size: 18px;
                  font-weight: 500;
                }

                .price-change {
                  font-size: 14px;
                  font-weight: 500;
                  margin-left: 8px;
                }

                .change-icon {
                  font-size: 16px;
                  margin-left: 8px;
                }

                &.change-up {
                  color: $green1;
                }

                &.change-same {
                  color: $silver;
                }

                &.change-down {
                  color: $red2;
                }
              }
            }
          }
        }

        .transaction-list {
          margin: 0 29px 20px 0;

          .transaction-list-header {
            h2 {
              font-size: 18px;
              margin-bottom: 16px;
            }
          }

          .transaction-list-item {
            border-radius: 8px;
            box-sizing: content-box;
            display: flex;
            padding: 16px;

            .transaction-icon {
              align-items: center;

              border-radius: 50%;

              display: flex;
              font-size: 16px;
              height: 32px;
              justify-content: center;
              margin-right: 10px;
              width: 32px;

              @include themify(day) {
                background: $white;
                color: $steel-grey;
              }

              @include themify(night) {
                background: $metallic-blue;
                color: lighten($metallic-blue, 40);
              }
            }

            .transaction-title {
              display: flex;
              flex-direction: column;
              font-weight: 500;
              width: 30%;

              .transaction-name {
                @include themify(day) {
                  color: $steel-grey;
                }

                @include themify(night) {
                  color: $metallic-blue;
                }
              }

              .transaction-date {
                font-size: 12px;

                @include themify(day) {
                  color: $silver;
                }

                @include themify(night) {
                  color: $warm-grey;
                }
              }
            }

            .transaction-numbers {
              color: $dark-sky-blue;
              flex-grow: 1;

              font-weight: 500;
              margin-right: 30px;
              text-align: right;

              .number {
                margin-right: 4px;
              }
            }

            .transaction-details {
              color: $steel-grey;
              width: 30%;
              word-break: break-all;
            }

            &:nth-child(odd) {
              @include themify(day) {
                background: $white-three;
              }

              @include themify(night) {
                background: lighten($dark-six, 4);
              }
            }

            @include themify(day) {
              &:nth-child(even) {
                .transaction-icon {
                  background-color: $white-three;
                }
              }
            }
          }
        }
      }
    }
  }
}
