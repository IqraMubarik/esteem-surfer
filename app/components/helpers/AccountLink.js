import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { getAccount } from '../../backend/steem-client';

export const makePath = username => `/@${username}`;

class AccountLink extends Component {
  goProfile = async () => {
    const { username, history, actions } = this.props;
    let { accountData } = this.props;
    const { setVisitingAccount } = actions;

    if (!accountData) {
      try {
        accountData = await getAccount(username);
      } catch (err) {
        accountData = null;
      }
    }

    if (accountData) {
      setVisitingAccount(accountData);
    }

    history.push(makePath(username));
  };

  render() {
    const { children } = this.props;

    return React.cloneElement(children, {
      onClick: this.goProfile
    });
  }
}

AccountLink.defaultProps = {
  accountData: null
};

AccountLink.propTypes = {
  children: PropTypes.element.isRequired,
  username: PropTypes.string.isRequired,
  history: PropTypes.instanceOf(Object).isRequired,
  accountData: PropTypes.instanceOf(Object),
  actions: PropTypes.shape({
    setVisitingAccount: PropTypes.func.isRequired
  }).isRequired
};

export default AccountLink;
