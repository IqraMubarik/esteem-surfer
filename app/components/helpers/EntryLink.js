import React, {Component} from 'react';
import PropTypes from 'prop-types';

class EntryLink extends Component {
  goEntry = () => {
    const {entry, history, actions} = this.props;

    const {category, author, permlink} = entry;

    actions.setVisitingEntry(entry);

    history.push(`/${category}/@${author}/${permlink}`);
  };

  render() {
    const {children} = this.props;

    return React.cloneElement(children, {
      onClick: this.goEntry
    });
  }
}

EntryLink.defaultProps = {};

EntryLink.propTypes = {
  children: PropTypes.element.isRequired,
  history: PropTypes.instanceOf(Object).isRequired,
  entry: PropTypes.shape({
    category: PropTypes.string.isRequired,
    author: PropTypes.string.isRequired,
    permlink: PropTypes.string.isRequired
  }).isRequired,
  actions: PropTypes.shape({
    setVisitingEntry: PropTypes.func.isRequired
  }).isRequired
};

export default EntryLink;
