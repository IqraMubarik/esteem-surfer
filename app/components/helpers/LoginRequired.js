import React, {Component, Fragment} from 'react';
import PropTypes from "prop-types";
import Login from '../dialogs/Login'
import {Modal} from "antd";


class LoginRequired extends Component {


  constructor(props) {
    super(props);

    this.state = {
      loginModalVisible: false
    };
  }


  showLoginModal = () => {

    console.log("aaa")
    this.setState({
      loginModalVisible: true
    });
  };

  onLoginModalCancel = () => {
    this.setState({
      loginModalVisible: false
    });
  };

  onLoginSuccess = () => {
    this.setState({
      loginModalVisible: false
    });
  };

  render() {
    const {children, activeAccount, requiredKeys} = this.props;
    const {loginModalVisible} = this.state;

    if (activeAccount) {
      return children
    }


    const newChild = React.cloneElement(children, {
      onClick: () => {
        this.showLoginModal()
      }
    });

    return (
      <Fragment>
        {newChild}

        <Modal
          visible={loginModalVisible}
          onCancel={this.onLoginModalCancel}
          footer={false}
          width="500px"
          closable={false}
          destroyOnClose
          centered
        >
          <Login {...this.props} onSuccess={this.onLoginSuccess}/>
        </Modal>
      </Fragment>
    )
  }
}

LoginRequired.defaultProps = {
  activeAccount: null,
};


LoginRequired.propTypes = {
  children: PropTypes.element.isRequired,
  activeAccount: PropTypes.instanceOf(Object),
  requiredKeys: PropTypes.array.isRequired
};

export default LoginRequired;
