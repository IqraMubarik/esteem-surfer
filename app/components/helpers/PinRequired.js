import React, { Component, Fragment } from 'react';
import { Modal } from 'antd';

import PropTypes from 'prop-types';
import PinConfirm from '../dialogs/PinConfirm';


class PinRequired extends Component {

  constructor(props) {
    super(props);

    this.state = {
      show: false
    };
  }

  onPinInvalidated = () => {

  };

  onConfirmPinSuccess = () => {
    const { children } = this.props;
    const { onClick } = children.props;
    this.setState({ show: false }, () => {
      onClick();
    });
  };

  render() {
    const { children } = this.props;
    const { show } = this.state;

    // Clone child element rewriting onclick event
    const newChild = React.cloneElement(children, {
      onClick: () => {
        this.setState({ show: true });
      }
    });

    return (
      <Fragment>
        {show &&
        <Modal
          footer={null}
          closable={false}
          keyboard={false}
          visible
          width="500px"
          centered
          destroyOnClose
        >
          <PinConfirm
            {...this.props}
            onSuccess={this.onConfirmPinSuccess}
            onInvalidate={this.onPinInvalidated}
          />
        </Modal>
        }
        {newChild}
      </Fragment>
    );
  }
}

PinRequired.defaultProps = {};

PinRequired.propTypes = {
  children: PropTypes.element.isRequired
};

export default PinRequired;
