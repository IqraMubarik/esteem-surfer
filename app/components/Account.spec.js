/* eslint-disable */

import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { AccountMenu, AccountCover, AccountTopPosts } from './Account';
import { mountWithIntl, shallowWithIntl } from '../helpers/intl-test';

Enzyme.configure({ adapter: new Adapter() });

describe('Account', () => {
  it('(1) AccountMenu render', () => {
    const props = {
      username: 'foo',
      section: 'comments',
      history: {},
      global: {
        listStyle: 'row'
      },
      actions: {
        changeListStyle: () => {}
      }
    };
    const wrapper = mountWithIntl(<AccountMenu {...props} />);

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('(2) AccountCover render', () => {
    const props = {
      username: 'foo',
      account: {
        accountProfile: {
          cover_image: 'https://foo.bar/baz.png'
        }
      },
      global: {
        theme: 'day'
      }
    };
    const wrapper = mountWithIntl(<AccountCover {...props} />);

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('(3) AccountTopPosts render', () => {
    const props = {
      posts: [
        {
          title: 'Foo bar',
          body: 'lorem ipsum',
          permlink: 'lorem-ipsum'
        },
        {
          title: 'Baz gaz',
          body: 'dolor sit amet',
          permlink: 'dolor-sit'
        }
      ]
    };
    const wrapper = mountWithIntl(<AccountTopPosts {...props} />);

    expect(wrapper.html()).toMatchSnapshot();
  });
});
