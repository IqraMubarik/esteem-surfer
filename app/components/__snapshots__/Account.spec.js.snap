// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account (1) AccountMenu render 1`] = `"<div class=\\"account-menu\\"><div class=\\"account-menu-items\\"><a role=\\"none\\" class=\\"menu-item false\\"><span>Blog</span></a><a role=\\"none\\" class=\\"menu-item selected-item\\"><span>Comments</span></a><a role=\\"none\\" class=\\"menu-item false\\"><span>Replies</span></a><a role=\\"none\\" class=\\"menu-item false\\"><span>Wallet</span></a></div><div class=\\"page-tools\\"><a class=\\"list-switch \\" role=\\"none\\"><i class=\\"mi\\">view_module</i></a></div></div>"`;

exports[`Account (2) AccountCover render 1`] = `"<div class=\\"account-cover\\"><div class=\\"cover-image\\" style=\\"background-image: url(https://steemitimages.com/0x0/https://foo.bar/baz.png);\\"></div><div class=\\"follow-controls-holder\\"><div><button style=\\"margin-right: 5px;\\" type=\\"button\\" class=\\"ant-btn ant-btn-primary\\"><span>Follow</span></button><button type=\\"button\\" class=\\"ant-btn\\"><span>Mute</span></button></div></div></div>"`;

exports[`Account (3) AccountTopPosts render 1`] = `"<div class=\\"top-posts-list\\"><h2 class=\\"top-posts-list-header\\"><span>Top Posts</span></h2><div class=\\"top-posts-list-body\\"><div class=\\"top-posts-list-item\\"><div class=\\"post-image\\"><img alt=\\"\\" src=\\"img/noimage.png\\"></div><div class=\\"post-content\\"><div class=\\"post-title\\">Foo bar</div><div class=\\"post-body\\">lorem ipsum</div></div></div><div class=\\"top-posts-list-item\\"><div class=\\"post-image\\"><img alt=\\"\\" src=\\"img/noimage.png\\"></div><div class=\\"post-content\\"><div class=\\"post-title\\">Baz gaz</div><div class=\\"post-body\\">dolor sit amet</div></div></div></div></div>"`;
